metrics/collector_test.go

```diff
@@ -121,6 +121,52 @@ func testMachineStatus(t *testing.T) {
 			if !found {
 				t.Error("metrics sabakan_machine_status was not found")
 			}
+
+			// If machines are deleted, corresponding metrics is also deleted
+			err = model.Machine.SetState(context.Background(), "001", sabakan.StateRetiring)
+			if err != nil {
+				t.Fatal(err)
+			}
+			err = model.Machine.SetState(context.Background(), "001", sabakan.StateRetired)
+			if err != nil {
+				t.Fatal(err)
+			}
+			err = model.Machine.Delete(context.Background(), "001")
+			if err != nil {
+				t.Fatal(err)
+			}
+			err = model.Machine.SetState(context.Background(), "002", sabakan.StateRetiring)
+			if err != nil {
+				t.Fatal(err)
+			}
+			err = model.Machine.SetState(context.Background(), "002", sabakan.StateRetired)
+			if err != nil {
+				t.Fatal(err)
+			}
+			err = model.Machine.Delete(context.Background(), "002")
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			w = httptest.NewRecorder()
+			req = httptest.NewRequest("GET", "/metrics", nil)
+			handler.ServeHTTP(w, req)
+			metricsFamily, err = parseMetrics(w.Result())
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			found = false
+			for _, mf := range metricsFamily {
+				if *mf.Name == "sabakan_machine_status" {
+					found = true
+					break
+				}
+			}
+			if found {
+				t.Error("metrics sabakan_machine_status was not deleted")
+			}
```

Comment:
You should not write the test here.
This is a table test, do you need to run your test for each test case?

Please create a new test function.